### PR TITLE
Staging to prod

### PR DESCRIPTION
**Pull request by Jaszkowic**

### CODE REVIEW

**Comment by Jaszkowic** [comment]
This can be merged to prod independently, as it does not require any Frontend changes. It will prepare the frontend to be ready for a) contact request feature and b) Gdk Stats feature.

File: supabase/functions/_shared/checks.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	recipientContactName: string,
+	token: string,
+	supabaseClient: SupabaseClient,
+	supabaseServiceRoleClient: SupabaseClient
+): Promise<CheckResult> {
+	// Get the user (= sender) data from the token
+	const { data: senderData, error: senderDataError } =
+		await supabaseClient.auth.getUser(token);
+
+	console.log(senderData);
```
Comment (ff6347) [inline supabase/functions/_shared/checks.ts:27]: Should we lo this?

File: supabase/functions/_shared/checks.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+	// Lookup the sender username
+	const { data: senderLookupData, error: senderLookupDataError } =
+		await supabaseServiceRoleClient
+			.from("profiles")
+			.select("*")
+			.eq("id", senderData.user.id)
+			.single();
+
+	console.log(senderLookupData);
```
Comment (ff6347) [inline supabase/functions/_shared/checks.ts:42]: Should we log this?

File: supabase/functions/_shared/checks.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	const { data: requestsToRecipient, error: requestsToRecipientError } =
+		await supabaseClient
+			.from("contact_requests")
+			.select("*")
+			.eq("user_id", senderData.user.id)
+			.eq("contact_id", recipientData.id)
+			.not("contact_mail_id", "is", null); // only count sent emails
+
+	if (requestsToRecipientError) {
+		console.log(requestsToRecipientError);
```
Comment (ff6347) [inline supabase/functions/_shared/checks.ts:72]: console.error?

File: supabase/functions/_shared/checks.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	const { data: requestsOfLast24h, error: requestsOfLast24hError } =
+		await supabaseClient
+			.from("contact_requests")
+			.select("*")
+			.eq("user_id", senderData.user.id)
+			.not("contact_mail_id", "is", null) // only count sent emails
+			.gt("created_at", sub(new Date(), { days: 1 }).toISOString());
+
+	if (requestsOfLast24hError) {
+		console.log(requestsOfLast24hError);
```
Comment (ff6347) [inline supabase/functions/_shared/checks.ts:98]: console.error

File: supabase/functions/_shared/checks.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	if (senderLookupDataError) {
+		console.log(senderLookupDataError);
+		return { isAllowed: false, reason: "not_found", lookupData: undefined };
+	}
+
+	// Lookup the recipient user id
+	const { data: recipientData, error: recipientDataError } =
+		await supabaseServiceRoleClient
+			.from("profiles")
+			.select("*")
```
Comment (ff6347) [inline supabase/functions/_shared/checks.ts:53]: Could we be more sparse which fields we select?

File: supabase/functions/_shared/checks.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+			.eq("id", senderData.user.id)
+			.single();
+
+	console.log(senderLookupData);
+
+	if (senderLookupDataError) {
+		console.log(senderLookupDataError);
```
Comment (ff6347) [inline supabase/functions/_shared/checks.ts:45]: console.error

File: supabase/functions/check_contact_request/index.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import { createClient } from "https://esm.sh/@supabase/supabase-js@2";
+import { checkIfContactRequestIsAllowed } from "../_shared/checks.ts";
+import { corsHeaders } from "../_shared/cors.ts";
+
+const SUPABASE_URL = Deno.env.get("SUPABASE_URL");
+const SUPABASE_ANON_KEY = Deno.env.get("SUPABASE_ANON_KEY");
+const SUPABASE_SERVICE_ROLE_KEY = Deno.env.get("SUPABASE_SERVICE_ROLE_KEY");
+
+const handler = async (_request: Request): Promise<Response> => {
```
Comment (ff6347) [inline supabase/functions/check_contact_request/index.ts:9]: No error handling at all?

File: supabase/functions/check_contact_request/index.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
+import { createClient } from "https://esm.sh/@supabase/supabase-js@2";
+import { checkIfContactRequestIsAllowed } from "../_shared/checks.ts";
+import { corsHeaders } from "../_shared/cors.ts";
+
+const SUPABASE_URL = Deno.env.get("SUPABASE_URL");
```
Comment (ff6347) [inline supabase/functions/check_contact_request/index.ts:5]: No check if these vars don't exisit

File: supabase/functions/check_contact_request/index.ts (hunk 7 above)
Comment (ff6347) [inline supabase/functions/check_contact_request/index.ts:9]: Why _request?

File: supabase/functions/submit_contact_request/index.ts [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+const SMTP_PASSWORD = Deno.env.get("SMTP_PASSWORD");
+const SMTP_FROM = Deno.env.get("SMTP_FROM");
+const SMTP_PORT = parseInt(Deno.env.get("SMTP_PORT"));
+const SMTP_SECURE = Deno.env.get("SMTP_SECURE") === "true";
+
+const SUPABASE_URL = Deno.env.get("SUPABASE_URL");
+const SUPABASE_ANON_KEY = Deno.env.get("SUPABASE_ANON_KEY");
+const SUPABASE_SERVICE_ROLE_KEY = Deno.env.get("SUPABASE_SERVICE_ROLE_KEY");
+
+const handler = async (_request: Request): Promise<Response> => {
```
Comment (ff6347) [inline supabase/functions/submit_contact_request/index.ts:18]: No error handling

File: supabase/functions/submit_contact_request/index.ts [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+
+const SMTP_HOST = Deno.env.get("SMTP_HOST");
+const SMTP_USER = Deno.env.get("SMTP_USER");
+const SMTP_PASSWORD = Deno.env.get("SMTP_PASSWORD");
+const SMTP_FROM = Deno.env.get("SMTP_FROM");
+const SMTP_PORT = parseInt(Deno.env.get("SMTP_PORT"));
+const SMTP_SECURE = Deno.env.get("SMTP_SECURE") === "true";
+
+const SUPABASE_URL = Deno.env.get("SUPABASE_URL");
+const SUPABASE_ANON_KEY = Deno.env.get("SUPABASE_ANON_KEY");
```
Comment (ff6347) [inline supabase/functions/submit_contact_request/index.ts:15]: No checks for env existence

File: supabase/migrations/20240604135436_contact_requests.sql [hunk 11]
```diff
@@ -0,0 +1,49 @@
+create table contact_requests (
+  id uuid primary key default gen_random_uuid(),
+  user_id uuid references auth.users(id) not null,
+  contact_id uuid references auth.users(id) not null,
+  created_at timestamp with time zone default timezone('utc'::text, now()) not null,
+  contact_message text,
+  contact_mail_id text default null -- the resend.io ID of the sent contact mail
```
Comment (ff6347) [inline supabase/migrations/20240604135436_contact_requests.sql:7]: Is this resend.io thing still relevant?

**Review by ff6347 — CHANGES_REQUESTED** [review]
Might be nitpicking but I'm missing some basic error handling. Many async request that might fail and are not handled. Also there are some points with unneeded console.logs that seem to be leftover from debugging.

File: supabase/functions/check_contact_request/index.ts (hunk 8 above)
Comment (Jaszkowic) [reply supabase/functions/check_contact_request/index.ts]: implemented `loadEnvVars` to check for that

File: supabase/functions/submit_contact_request/index.ts (hunk 9 above)
Comment (Jaszkowic) [reply supabase/functions/submit_contact_request/index.ts]: implented error handling in later commits

File: supabase/functions/submit_contact_request/index.ts (hunk 10 above)
Comment (Jaszkowic) [reply supabase/functions/submit_contact_request/index.ts]: implemented `loadEnvVars` to check for that

**Pull request merged**